File: app/src/main/java/de/christinecoenen/code/zapp/app/player/PersistedPlaybackPositionRepository.java
package de.christinecoenen.code.zapp.app.player;

import de.christinecoenen.code.zapp.app.mediathek.repository.MediathekRepository;
import io.reactivex.Single;

public class PersistedPlaybackPositionRepository implements IPlaybackPositionRepository {

	private final MediathekRepository mediathekRepository;

	public PersistedPlaybackPositionRepository(MediathekRepository mediathekRepository) {
		this.mediathekRepository = mediathekRepository;
	}

	@Override
	public void savePlaybackPosition(VideoInfo videoInfo, long millis) {
		if (videoInfo.getId() == 0) {
			return;
		}

		mediathekRepository.setPlaybackPosition(videoInfo.getId(), millis);
	}

	@Override
	public Single<Long> getPlaybackPosition(VideoInfo videoInfo) {
		if (videoInfo.getId() == 0) {
			return Single.just(0L);
		}

		return mediathekRepository.getPlaybackProsition(videoInfo.getId());
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/app/player/PersistedPlaybackPositionRepository.kt
package de.christinecoenen.code.zapp.app.player

import de.christinecoenen.code.zapp.repositories.MediathekRepository

class PersistedPlaybackPositionRepository(
	private val mediathekRepository: MediathekRepository
) : IPlaybackPositionRepository {

	override suspend fun savePlaybackPosition(
		videoInfo: VideoInfo,
		positionMillis: Long,
		durationMillis: Long
	) {
		if (videoInfo.id == 0) {
			return
		}

		mediathekRepository.setPlaybackPosition(videoInfo.id, positionMillis, durationMillis)
	}

	override suspend fun getPlaybackPosition(videoInfo: VideoInfo): Long {
		return if (videoInfo.id == 0) {
			0L
		} else {
			mediathekRepository.getPlaybackPosition(videoInfo.id)
		}
	}

}
